File: Medium/1833_Maximum_Ice_Cream_Bars/Solution.cpp
#include "../../devlibs/cpp/cpphelper.h"

using namespace std;


// NOTE:
// For LeetCode submission, copy only the `class Solution` part.


class Solution {
public:
    int maxIceCream(vector<int>& costs, int coins) {
        sort(costs.begin(), costs.end());

        int count = 0;

        for (int cost : costs) {
            if (coins < cost) break;

            coins -= cost;
            count++;
        }

        return count;
    }
};


int main(int argc, char* argv[]) {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    Solution solution;

    // test cases 1
    vector<int> costs1 = {1,3,2,4,1};
    int coins1 = 7;
    cout << solution.maxIceCream(costs1, coins1) << endl;

    // test cases 2
    vector<int> costs2 = {10,6,8,7,7,8};
    int coins2 = 5;
    cout << solution.maxIceCream(costs2, coins2) << endl;

    // test cases 3
    vector<int> costs3 = {1,6,3,1,2,5};
    int coins3 = 20;
    cout << solution.maxIceCream(costs3, coins3) << endl;

    return 0;
}

File: Medium/1833_Maximum_Ice_Cream_Bars/Solution.py
from typing import List, Optional


class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        costs.sort()

        count = 0

        for cost in costs:
            if (coins < cost):
                break

            coins -= cost
            count += 1
        
        return count


if __name__ == "__main__":
    solution = Solution()

    # test cases 1
    costs1 = [1,3,2,4,1]
    coins1 = 7
    print(solution.maxIceCream(costs1, coins1))

    # test cases 2
    costs2 = [10,6,8,7,7,8]
    coins2 = 5
    print(solution.maxIceCream(costs2, coins2))

    # test cases 3
    costs3 = [1,6,3,1,2,5]
    coins3 = 20
    print(solution.maxIceCream(costs3, coins3))
    
    